### PR TITLE
add clear explanation of what the plugin does

### PR DESCRIPTION
**Pull request by zwhitchcox**
So, for a long time, I went without really knowing what this plugin does and have had a lot of errors and frustration with rollup, because of it. I knew I was supposed to use it for certain plugins, but had no idea why. I read the docs, but was still clueless as to what the plugin actually did. I think I finally figured it out and decided to share this knowledge with everyone else by adding a more in depth explanation with examples to be user friendly for people who might not understand the intricacies of rollup like me.

Still not *totally* sure that this is what it does, but it's what it seems like to me. In any case, if it's not, I think that's beside the point, and we should try to make it easy for people to understand.

Wow, that was more long winded than I intended.

### CODE REVIEW

**Comment by lukastaegert** [comment]
Hi @zwhitchcox, thanks for this PR! This really shows us that it would really be helpful if the documentation was improved here.

However your description is not entirely correct so let me clarify things a little and maybe we can work towards a better example.

Traditionally, node modules specify imports and exports via the CommonJS format. In this format, modules are imported via `require()` and variables are exported by attaching them to the `exports` object (or by assigning to `module.exports` for node specifically, though strictly speaking, this is not part of the CJS specification).

Rollup, on the other hand, is built around and only understands the ECMAScript module format (which is sort of the `official` JavaScript module format now). Here, modules are imported via `import "my-module"` or `import xyz from "my-module"` and variables are exported via `export const x = 2` or `export { x }`.

The purpose of this plugin is basically to transform the CJS format to the ESM format (so it is a little different from your example). To get an idea what this looks like, look at the input and output of the tests here: https://github.com/rollup/rollup-plugin-commonjs/tree/master/test/form

Now the question is how to best reflect this in a readme concisely but in a way that people not that familiar with these concepts will quickly understand why and when they need this.

File: README.md [hunk 1]
```diff
@@ -5,6 +5,32 @@
...
+import log2 from 'log2' // [!] Error: 'default' is not exported  by node_modules/log2/index.js
+```
+
+However, with `rollup-plugin-commonjs`, the code becomes
+
+```js
+module.exports.default = function log2() {console.log(2)}
+```
+
+and rollup can now the module, and you can log the number 2!
```
Comment (probablyup) [inline README.md]: and rollup can now the module -> and rollup can now __consume__ the module

**Comment by shrpne** [comment]
@lukastaegert I think your description is great and could be added to the readme as is.

**Comment by shellscape** [comment]
I'll personally oversee a merge between the descriptions and information in this PR into the documentation for the plugin once we move it to a new home at `rollup/plugins`. Thanks for opening this PR and starting the discussion

**Pull request closed without merging**